Userland/Libraries/LibWeb/Geometry/DOMMatrix.cpp

```diff
@@ -252,11 +252,14 @@ void DOMMatrix::set_f(double value)
 // https://drafts.fxtf.org/geometry/#dom-dommatrix-invertself
 JS::NonnullGCPtr<DOMMatrix> DOMMatrix::invert_self()
 {
+    bool is_invertible = m_matrix.is_invertible();
+
     // 1. Invert the current matrix.
-    m_matrix = m_matrix.inverse();
+    if (is_invertible)
+        m_matrix = m_matrix.inverse();
 
     // 2. If the current matrix is not invertible set all attributes to NaN and set is 2D to false.
-    if (!m_matrix.is_invertible()) {
+    if (!is_invertible) {
```

Comment:
why not use `if-else` here?

Reply:
Because now it more accurately reflects the spec. What would be gained from an `else` here?

Reply:
just easier to read. 🙂